backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -448,14 +448,18 @@ def grant_dataset_bucket_key_policy(self):
                 }
             kms_client.put_key_policy(kms_key_id, json.dumps(existing_policy))
 
-    def delete_target_role_bucket_policy(self):
+    def delete_target_role_bucket_policy(self, principal_exist):
         logger.info(f'Deleting target role from bucket policy for bucket {self.bucket_name}...')
         try:
             s3_client = S3Client(self.source_account_id, self.source_environment.region)
             bucket_policy = json.loads(s3_client.get_bucket_policy(self.bucket_name))
-            target_requester_arn = IAM.get_role_arn_by_name(
-                self.target_account_id, self.target_environment.region, self.target_requester_IAMRoleName
-            )
+            if principal_exist:
```

Comment:
I don't think we need to pass the `principal_exist`, we can just use the function `IAM.get_role_arn_by_name` if it returns `None` then the principal does not exist. Like what is implemented in line 549

Reply:
adressed in [b60f12d](https://github.com/data-dot-all/dataall/pull/1161/commits/b60f12d84c0cef198310dc6b022753d925816b7c)